interface/src/app/analyze/analysis/sampleBin.js

```diff
@@ -183,10 +197,86 @@ function($log, $cacheFactory, $q, Sample, Activity, NodeInfo, MathFuncts) {
       ).$promise;
       return pSample;
     },
+    getCachedNodeInfo: function(pk) {
+      return this.nodeCache.get(pk);
+    },
+    getNodeInfoSetPromise: function(pkArr) {
+      // Check for any pk already cached, then retrieve what's missing in
+      // bulk via the set endpoint on the node API. Return a promise and
+      // supply a callback that populates the cache when the API returns.
+      var cbSampleBin = this; // closure link to SampleBin for callbacks
```

Comment:
Exactly. This variable gives me a well-defined way for callback functions to access SampleBin properties and invoke its methods.